### DISCUSSION TITLE
Download from Attachment - action not working

### DISCUSSION THREAD

**Discussion by Rajaelango** [Budibase/budibase#13750] [Help]
I'm new to Budibase, Could someone guide how to set Download from Attachment - action?  below is my current setup however I'm not able to download the file for the selected row from my table. 

![Screenshot 2024-05-23 170617](https://github.com/Budibase/budibase/assets/48485992/9cb2392b-c4cc-4ea8-a941-bb8e5dcff36f)

**Comment by ConorWebb96** [comment]
Hey @Rajaelango,

You're missing the rowID, if you don't have access to that. You can click the select to the right of the file label and switch to URL. From here you would have to supply the url of the attachment itself.

More information on this can be found [here](https://docs.budibase.com/docs/data-actions#download-file).

**Reply by Rajaelango** [reply]
Hi @ConorWebb96 , Thanks for the reply. I would like to use the File as an attachment. 

I have added the rowID and I'm still facing the below error:

 **Failed to download from 'undefined'. The server returned status code: 404** .

My goal is to download the attachments for a selected row when click on the download button.  

![Screenshot 2024-05-24 142957](https://github.com/Budibase/budibase/assets/48485992/c5fe31e6-1b8d-41cd-b303-6012d551a369)

**Comment by mikesealey** [comment]
If the Download button isn't specific to the row, only the table, you might not be getting or setting the row ID correctly. If you're selecting row**s** and downloading from there, your row ID might actually be an array of row IDs, which would require a little more though.

I would suggest the first step to debugging this would be to set a Paragraph component to display the value stored in `{{ rqdbtable.rqdb._id }}` and see if it is what you're expecting it to be.
